api/src/main/java/org/bahmni/module/bahmni/ie/apps/service/impl/BahmniFormPrivilegesServiceImpl.java

```diff
@@ -59,54 +55,49 @@ public FormPrivilege saveFormPrivilege(FormPrivilege toPersistFormPrivilege) thr
         List<FormPrivilege> privilegesList = new ArrayList<FormPrivilege>();
         List<FormPrivilege> oldPrivilegeList = new ArrayList<FormPrivilege>();
 
-            Iterator privilegeItr = formPrivileges.iterator();
-            int formId = 0;
-            String formVersion = "";
-
-            while (privilegeItr.hasNext()) {
-                LinkedHashMap temp = (LinkedHashMap) privilegeItr.next();
-                List<Map.Entry> entrySetList = new ArrayList<Map.Entry>(temp.entrySet());
-                FormPrivilege tempPrivilege = new FormPrivilege();
-                for (Map.Entry tempEntrySet : entrySetList) {
-                    if (tempEntrySet.getKey().toString().equalsIgnoreCase("formId")) {
-                        tempPrivilege.setFormId((Integer) temp.get("formId"));
-                        formId = (Integer) temp.get("formId");
-                    } else if (tempEntrySet.getKey().toString().equalsIgnoreCase("editable")) {
-                        tempPrivilege.setEditable((Boolean) temp.get("editable"));
-                    } else if (tempEntrySet.getKey().toString().equalsIgnoreCase("privilegeName")) {
-                         String privilegeName = temp.get("privilegeName").toString();
-                         tempPrivilege.setPrivilegeName(privilegeName);
-                    } else if (tempEntrySet.getKey().toString().equalsIgnoreCase("viewable")) {
-                        tempPrivilege.setViewable((Boolean) temp.get("viewable"));
-                    }else if (tempEntrySet.getKey().toString().equalsIgnoreCase("formVersion")) {
-                        tempPrivilege.setFormVersion((String) temp.get("formVersion"));
-                        formVersion = tempPrivilege.getFormVersion();
-                    }
+        Iterator privilegeItr = formPrivileges.iterator();
+        int formId = 0;
+        String formVersion = "";
+
+        while (privilegeItr.hasNext()) {
+            LinkedHashMap temp = (LinkedHashMap) privilegeItr.next();
```

Comment:
Why are we using these temporary variables and looping through the entry?
If the purpose is to just prepare list of "FormPrivilege", just create an instance and copy the values 
1) Optional.ofNullable(temp.get("formId")).ifPresent(x -> { tempPrivilege.setFormId((Integer) x); });
or more conventional
2)  if (temp.get("formId") != null) {
         tempPrivilege.setFormId((Integer) temp.get("formId"));
     }

You don't need variables "formId" and "formVersion" as well. You can just get them "privilegesList.get(0).getFormId()" and "privilegesList.get(0).getFormVersion()" methods. Of course, you ought to make sure that "deleteAllThePrivilegesFromDB()" method checks that the variables are not null (and exit if so)

Reply:
@buvaneswari-arun are you going to work on this part?

Reply:
This is done.  The changes are committed.